nixos/modules/programs/direnv.nix

```
@@ -0,0 +1,112 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}: let
+  cfg = config.programs.direnv;
+in {
+  options.programs.direnv = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to enable direnv integration. Takes care of both installation and
+        setting up the sourcing of the shell. Additionally enables nix-direnv
+        integration. Note that you need to logout and login for this change to apply.
+      '';
+    };
+    direnvrcExtra = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        use_zig() {
+          zig_inst -q $1
+          ZIG_DIR="''${HOME}/.local/share/zig/$1"
+          ZIG_UPDATE="''${ZIG_DIR}/zig_update"
+          if [ ! -f "''${ZIG_UPDATE}" ]; then
+            echo -e "#!/bin/bash\nzig_inst -q -f $1\n" > "''${ZIG_UPDATE}"
+           chmod a+x "''${ZIG_UPDATE}"
+          fi
+          PATH_add "''${ZIG_DIR}"
+          PATH_add "./zig-cache/bin"
+          PATH_add "./zig-out/bin"
+          echo "Using zig $1"
+        }
+      '';
+      description = ''
+        Extra lines to append to the sourced direnvrc
+      '';
+    };
+    silent = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to hide direnv logging
+      '';
+    };
+    persistDerivations = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      example = false;
+      description = ''
+        Whether to set keep-derivations and keep-outputs to true
+        to prevent shells from getting garbage collected
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    programs = {
+      zsh.interactiveShellInit = ''
+        eval "$(direnv hook zsh)"
+      '';
+      bash.interactiveShellInit = ''
+        eval "$(direnv hook bash)"
+      '';
+      fish.interactiveShellInit = ''
+        direnv hook fish | source;
+      '';
+    };
+    nix.settings = lib.mkIf cfg.persistDerivations {
+      keep-outputs = true;
+      keep-derivations = true;
+    };
+    environment = {
+      systemPackages = [pkgs.direnv];
```

```suggestion
      systemPackages = with pkgs; [ direnv ];
```
don't we need nix-direnv? https://github.com/NixOS/nixpkgs/pull/192667/files#diff-ff3b3a796d514f9f77a780213ec8446daf34394c6b097f837cc226231b21af30R16-R19

---

No need with this strategy it can just be sourced directly in the direnvrc

---

Oh! I accidentally deleted the sourcing on the last push

---

To be clear: It's already in the direnvrc. We are referencing it in `/etc/direnv/direnvrc` (see 108 below)